apm-agent-plugins/apm-scala-concurrent-plugin/src/test/scala/co/elastic/apm/agent/scala/concurrent/FutureInstrumentationSpec.scala

```diff
@@ -0,0 +1,91 @@
+package co.elastic.apm.agent.scala.concurrent
+
+import java.util.concurrent.Executors
+
+import co.elastic.apm.agent.MockReporter
+import co.elastic.apm.agent.bci.ElasticApmAgent
+import co.elastic.apm.agent.configuration.{CoreConfiguration, SpyConfiguration}
+import co.elastic.apm.agent.impl.transaction.Transaction
+import co.elastic.apm.agent.impl.{ElasticApmTracer, ElasticApmTracerBuilder}
+import net.bytebuddy.agent.ByteBuddyAgent
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.stagemonitor.configuration.ConfigurationRegistry
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext, ExecutionContextExecutor, Future}
+
+class FutureInstrumentationSpec extends AnyFlatSpec with Matchers with BeforeAndAfterEach {
+
+  implicit def executionContext: ExecutionContextExecutor =
+    ExecutionContext.fromExecutor(Executors.newFixedThreadPool(1))
+
+  private var reporter: MockReporter = _
+  private var tracer: ElasticApmTracer = _
+  private var coreConfiguration: CoreConfiguration = _
+  private var transaction: Transaction = _
+
+  override def beforeEach: Unit = {
+    reporter = new MockReporter
+    val config: ConfigurationRegistry = SpyConfiguration.createSpyConfig
+    coreConfiguration = config.getConfig(classOf[CoreConfiguration])
+    tracer = new ElasticApmTracerBuilder().configurationRegistry(config).reporter(reporter).build
+    ElasticApmAgent.initInstrumentation(tracer, ByteBuddyAgent.install)
+    transaction = tracer.startRootTransaction(null).withName("Transaction").activate()
+  }
+
+  override def afterEach: Unit = {
+    transaction.deactivate().end()
+    ElasticApmAgent.reset()
+  }
+
+  "test" should "test" in {
+    new TestFutureTraceMethods().invokeAsync()
```

Comment:
I see the FutureInstrumentation match correctly on this method-call which I find good progress :)

Im confused on why the test is not passing due to not having any transactions in the reporter although I manually create that root-transaction.
Any reason why it is not getting this root-transaction that gets created before every test?

Reply:
That is because you are ending the transaction after the test has run (`afterEach` runs after `"test"`). Transactions are reported only after they are ended.